pandas/tests/io/formats/test_to_csv.py

```diff
@@ -561,3 +561,19 @@ def test_to_csv_compression(self, compression_only,
             result = pd.read_csv(path, index_col=0,
                                  compression=read_compression)
             tm.assert_frame_equal(result, df)
+
+    def test_to_csv_na_rep_long_string(self, capsys):
+        # see gh-25099
+        df = pd.DataFrame({"c": [float('nan')] * 3})
+        df = df.astype("Int64")
+        expected_rows = ['c',
+                         'mynull',
+                         'mynull',
+                         'mynull']
+        expected_ascii = tm.convert_rows_list_to_csv_str(expected_rows)
+
```

Comment:
just write to a StringIO, no need to use stdout here.

use result= and expected=

Reply:
Sure thing! Updating the branch in a sec. Thanks @jreback !

Reply:
Done!